<?xml version="1.0"?>	
 <launch>	
 	<arg name="sim" default="false"/>	

	<!-- Start Grasp Planning node -->	
    <node name="grasp_planner" pkg="simple_grasping" type="grasp_planner_node" output="screen">	
    	<param name="simulated_kinova" value="$(arg sim)"/>	
        <rosparam command="load" file="$(find movo_demos)/config/simple_grasping.yaml"/>	

        <rosparam if="$(optenv MOVO_HAS_KINOVA_ARM_6DOF false)" command="load" file="$(find movo_moveit_config)/config/kinematics.yaml"/>	
        <rosparam if="$(optenv MOVO_HAS_KINOVA_ARM_7DOF false)" command="load" file="$(find movo_7dof_moveit_config)/config/kinematics.yaml"/>	
    </node>	

    <!-- Launch pick and place if 6 DOF -->
	<node if="$(optenv MOVO_HAS_KINOVA_ARM_6DOF false)" name="pick_place_demo" pkg="movo_demos" type="Alex_pick_place_demo.py" output="screen">	
    	<rosparam command="load" file="$(find movo_moveit_config)/config/kinematics.yaml"/>	
		<param name="sim" value="$(arg sim)"/>
		<param name="jaco_dof" value="6dof"/>
	</node>	

	<node if="$(optenv MOVO_HAS_RIGHT_ARM_6DOF false)" name="pick_place_demo" pkg="movo_demos" type="Alex_pick_place_demo.py" output="screen">	
    	<rosparam command="load" file="$(find movo_moveit_config)/config/kinematics.yaml"/>	
		<param name="sim" value="$(arg sim)"/>
		<param name="jaco_dof" value="6dof"/>
	</node>	

    <!-- Launch pick and place if 7 DOF -->
	<node if="$(optenv MOVO_HAS_KINOVA_ARM_7DOF false)" name="pick_place_demo" pkg="movo_demos" type="Alex_pick_place_demo.py" output="screen">	
    	<rosparam command="load" file="$(find movo_7dof_moveit_config)/config/kinematics.yaml"/>	
		<param name="sim" value="$(arg sim)"/>
		<param name="jaco_dof" value="7dof"/>
	</node>	

	<node if="$(optenv MOVO_HAS_LEFT_ARM_7DOF false)" name="pick_place_demo" pkg="movo_demos" type="Alex_pick_place_demo.py" output="screen">	
    	<rosparam command="load" file="$(find movo_moveit_config)/config/kinematics.yaml"/>	
		<param name="sim" value="$(arg sim)"/>
		<param name="jaco_dof" value="7dof"/>
	</node>	

</launch>	
